data_handler.py

```diff
@@ -133,8 +145,31 @@ def _append_additional_data(self, smiley_data: list) -> list:
             if row_rem != 0:
                 time.sleep(self.CRAWL_DELAY)
 
+        self.output_processed_companies(newly_processed_pnrs)
         return out
 
+    def output_processed_companies(self, restaurants: list):
+        file_exists = os.path.exists('processed_pnrs.csv')
+
+        with open('processed_pnrs.csv', 'a+') as f:
+            fieldnames = ['pnr', 'seneste_kontrol_dato']
+            writer = csv.DictWriter(
+                f, fieldnames=fieldnames, extrasaction='ignore')
+
+            if not file_exists:
+                writer.writeheader()
+
+            for restaurant in restaurants:
+                writer.writerow(restaurant)
+
+    def input_processed_companies(self) -> list:
```

Comment:
I stedet for at bruge en liste her som har O(N) lookup så brug en dictionary da den har O(1) lookup
contains_key kan evt være brugbart til det dette
`dictionary.contains_key( some_key )`

Reply:
👍
Jeg kigger på det

Reply:
Kan ikke se hvordan et dict skulle gøre forskel her? Hvert row er allerede en dict, og vi skal ikke accesse noget specifikt row, kun loope. Og vi har jo lige ændret hele strukturen fra dict til list for at gøre output mindre scuffed at arbejde med.